content/tutorials/build-a-data-source-backend-plugin.md

```diff
@@ -0,0 +1,115 @@
+
+{{< /tutorials/step >}}
+{{< tutorials/step duration="1" title="Create a new plugin" >}}
+
+To build a backend for your data source plugin, Grafana requires a binary that it can execute when it loads the plugin during start-up. In this guide, we will build a binary using our backend plugin SDK in Go.
+
+The easiest way to get started is to clone one of our test data datasources. Navigate to the plugin folder that you configured in step 1 and type:
+
+```
+npx @grafana/toolkit plugin:create my-plugin
+```
+
+select `Backend Datasource Plugin` and follow the rest of the steps in the plugin scaffolding command.
+
+```
+cd my-plugin
+```
+
+To build the plugin
+
+```
+yarn install --pure-lockfile
+ yarn build
+ mage -v buildAll
+```
+
+{{< /tutorials/step >}}
+{{< tutorials/step duration="1" title="Anatomy of a Backend Plugin" >}}
+
+The folders and files used to build the backend for the data source are
+
+| file/folder         | description                                                                                                                                        |
+| ------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `Magefile.go`       | It’s not a requirement to use mage build files, but we strongly recommend using it so that you can use the build steps provided in the plugin SDK. |
+| `/src/plugins.json` | A JSON file describing the backend plugin                                                                                                          |
+| `/pkg/main.go`      | Starting point of the plugin binary.                                                                                                               |
+
+#### plugin.json
+
+When building a backend plugin these fields are important:
+
+| field      | description                                                                                                            |
+| ---------- | ---------------------------------------------------------------------------------------------------------------------- |
+| backend    | Should be set to `true` for backend plugins. This tells Grafana that it should start a binary when loading the plugin. |
+| executable | This is the name of the executable that Grafana expects to start.                                                      |
+| alerting   | Should be set to true if your backend datasource supports alerting.                                                    |
+
+In the next step we will look at the query endpoint!
+
+{{< /tutorials/step >}}
+{{< tutorials/step duration="2" title="Implement data queries" >}}
+
+We begin by opening the file `/pkg/datasource/sample-datasource.go`. In this file you will see the `DatasourceInstance` struct which implements the `backend.QueryDataHandler` interface.
+The `QueryData` function on this struct is where the data fetching happens for a data source plugin.
```

Comment:
If a function is on a struct, it should by definition be a method (a function should be free-standing):
```suggestion
The `QueryData` method on this struct is where the data fetching happens for a data source plugin.
```